src/libraries/System.Text.Json/tests/Serialization/StreamTests.cs

```diff
@@ -0,0 +1,355 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class StreamTests
+    {
+        // To hit all possible continuation positions inside the tested object,
+        // the outer-class spacer needs to be between 5 and 116 bytes long.
+
+        // {"S":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","C":{"A":null,"B":"Hello","C":42,"D":null,"E":3.14E+17,"F":null,"G":true,"H":null,"I":[42,17],"J":{"A":null,"B":7}}}
+        // |<------------------------------------------------128 byte buffer------------------------------------------------------------->|
+        // {"S":"xxxxx","C":{"A":"Hello","B":null,"C":42,"D":null,"E":3.14E+17,"F":null,"G":true,"H":null,"I":[42,17],"J":{"A":null,"B":7}}}
+        public static IEnumerable<object[]> ContinuationSpacerLengths
+            => Enumerable.Range(5, 116 - 5 + 1).Select(length => new object[] { length });
```

Comment:
Please add some comments or constants on the 116 and 5 literals.

Once we get the first continuation, the buffer size will double to 256. With a small "S" string of 5 "xxxxx", length appears 128, so:
- To get continuation on the last EndObject tokens, don't we need 256 extra padding bytes?
- To get continuation on the first StartObject, don't we at least 1 extra padding byte?

Reply:
> Once we get the first continuation, the buffer size will double to 256.

Right now there is always only one continuation per test case. The padding determines where it happens.

- The 5 bytes padding leads to a first chunk that contains anything up to `...,"J":{"A":null,"B":7}}`. After continuation there is only the very last `EndObject` left.
- The 116 bytes padding leads to a fist chunk that ends with `...xxxxx","C":`. The continuation starts then with the `StartObject` of the tested object.

Maybe I don't get what you mean.

Reply:
OK thanks. I see now that there is only one continuation.